src/tcp_proxy.erl

```diff
@@ -136,26 +128,6 @@ terminate(_Reason, _State) ->
 code_change(_OldVsn, State, _Extra) ->
     {ok, State}.
 
--spec process_msgs([{msg, binary()} |
-                    {malformed_msg, binary()}], #state{}) -> #state{}.
-process_msgs(Msgs, S = #state{cb={Mod,ModState}}) ->
```

Comment:
You no-longer need the nested `#state.cb` functionality now (`logplex_worker` is dead, so you can drop it if there's only one callback, or replace the default with logplex_message otherwise). (See also `?DEFAULT_MSG_CB`)